soh/src/overlays/actors/ovl_En_Heishi4/z_en_heishi4.c

```diff
@@ -331,13 +332,38 @@ void func_80A56B40(EnHeishi4* this, PlayState* play) {
             return;
         }
         if (this->type == HEISHI4_AT_MARKET_NIGHT) {
+            if (CVar_GetS32("gRandoMarketSneak", 0)) {
+                this->actionFunc = EnHeishi4_MarketSneak;
+            } else {
             this->actionFunc = func_80A56614;
             return;
+            }
         }
     }
     func_8002F2F4(&this->actor, play);
 }
 
+/*
+Rando-specific hack function that allows child Link to exit from Market entrance to Hyrule Field
+at night.
+*/
+void EnHeishi4_MarketSneak(EnHeishi4* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == TEXT_STATE_CHOICE && Message_ShouldAdvance(play)) {
+        switch (play->msgCtx.choiceIndex) {
+            case 0: //yes
+                play->nextEntranceIndex = 0xCD;
```

Comment:
@Archez do we need to do anything special for entrance rando here? Or should this gracefully be handled correctly?

Reply:
Good question, I suppose the intent matters. In an overworld entrance rando, this would probably need to be wrapped with `play->nextEntranceIndex = Entrance_OverrideNextIndex(play->nextEntranceIndex)`

Reply:
> Good question, I suppose the intent matters. In an overworld entrance rando, this would probably need to be wrapped with `play->nextEntranceIndex = Entrance_OverrideNextIndex(play->nextEntranceIndex)`

So, `play->nextEntranceIndex = Entrance_OverrideNextIndex(0xCD)`? Just clarifying since none of this functionality exists in vanilla.